response_operations_ui/controllers/sample_controllers.py

```diff
@@ -28,6 +28,38 @@ def get_sample_summary(sample_summary_id):
     return response.json()
 
 
+def check_if_all_sample_units_present_for_sample_summary(sample_summary_id: str) -> bool:
+    """
+    Calls endpoint in sample that counts if expected sample units == actual number of sample units
+    and changes sample summary state to ACTIVE if the two number do match.
+
+    :param sample_summary_id:
+    :return:
+    """
+    logger.info("Checking sample summary state", sample_summary_id=sample_summary_id)
+    url = f'{app.config["SAMPLE_URL"]}/samples/samplesummary/{sample_summary_id}/check-all-units-present'
+    response = requests.get(url, auth=app.config["BASIC_AUTH"])
+
+    try:
+        response.raise_for_status()
+    except HTTPError:
```

Comment:
This does highlight the need to re-work the user journey around this process. An explicit button click "check sample load status" for example (rather than the page load) to transition the sample summary state, would be a nicer user experience. Presumably the CE page would throw a 500 if this exception was raised?

Reply:
Yeah that's fair.  I've just changed it so that it'll flash an error telling the user to refresh the page to try again.  Refreshing might not help every error, but if it's the error wasn't transient (like if the uuid for the sample summary was wrong for some reason) then we'd be in the same situation now needing manual intervention.